spec/components/provider_interface/application_choice_header_component_spec.rb

```diff
@@ -130,6 +130,40 @@
         allow(application_choice).to receive(:interviews).and_return(interviews)
       end
 
+      context 'when application is ended without success' do
+        let(:interviews) { class_double(Interview, kept: []) }
+
+        it 'does not render references tab' do
+          FeatureFlag.activate(:new_references_flow_providers)
+          %i[with_withdrawn_offer with_conditions_not_met with_rejection with_declined_offer].each do |factory|
+            application_choice = create(:application_choice, factory)
+            result = render_inline(
+              described_class.new(
+                application_choice:,
+              ),
+            )
+            expect(result.css('.app-tab-navigation').text).not_to include('References')
```

Comment:
Suggestion - could use the `have_link` matcher here, as the tabs are effectively just styled links?

```suggestion
            expect(rendered_component).to have_link('References')
```